### PR TITLE
added healthcheck.json

### PR DESCRIPTION
**Pull request by aldavidson**

### CODE REVIEW

File: postcodeinfo/apps/postcode_api/tests/views/test_support_views.py [hunk 1]
```diff
@@ -14,10 +15,11 @@ def setUp(self):
         token.save()
         self.valid_token = 'Token ' + str(self.user.auth_token)
 
+    # describe: ping.json
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/tests/views/test_support_views.py]: What's this?

File: postcodeinfo/apps/postcode_api/tests/views/test_support_views.py [hunk 2]
```diff
@@ -32,9 +34,65 @@ def test_that_getting_ping_json_with_no_token_responds_with_HTTP_200(self):
...
+                '/healthcheck.json', {}, True, False)
+        result = json.loads(response.content)
+        self.assertEqual( False, result['database']['ok'] )
+
+    def test_that_when_the_database_is_not_ok_it_shows_ok_false_at_the_top_level(self):
+        with patch('postcode_api.views.HealthcheckDotJsonView.is_database_ok', return_value=False):
+            response = self.client.get(
+                '/healthcheck.json', {}, True, False)
+        result = json.loads(response.content)
+        self.assertEqual( False, result['ok'] )
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/tests/views/test_support_views.py]: These test methods are quite copy-pasta. I think CodeClimate will complain. I think the similar forms can be distilled into some custom assert methods, eg:
```python
def request(self, path, token=None):
    headers = {}
    if token is not None:
        headers['HTTP_AUTHORIZATION'] = token
    return self.client.get(path, {}, True, False, **headers)

def healthcheck(self, **kwargs):
    return self.request('/healthcheck.json', **kwargs)

def assert_produces_parseable_json(self, response):
    try:
        json.loads(response.content)
    except ValueError:
        self.fail("Invalid JSON response: %s" % response.content)

def test_something_produces_parseable_json(self):
    self.assert_produces_parseable_json(self.healthcheck(token='Token 1234'))
```

File: postcodeinfo/apps/postcode_api/views.py [hunk 3]
```diff
@@ -79,3 +79,28 @@ def get(self, request, *args, **kwargs):
...
         return Response(data, status=status.HTTP_200_OK)
+
+
+class HealthcheckDotJsonView(generics.RetrieveAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def get(self, request, *args, **kwargs):
+        database_ok = self.is_database_ok()
+        # this should be an AND of all the checks - add more as needed
+        all_ok = database_ok
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/views.py]: The code would be self-explanatory if you use the `all` function here, eg:
```python
all_ok = all([
    self.is_database_ok(),
])
```

Comment (aldavidson) [inline postcodeinfo/apps/postcode_api/views.py]: nice - I've done it this way now. 

(I keep forgetting that the list methods are first-class functions in Python, I'm so used to having them as instance methods in Ruby!)

File: postcodeinfo/apps/postcode_api/tests/views/test_support_views.py [hunk 4]
```diff
@@ -25,16 +44,57 @@ def test_that_getting_ping_json_with_a_valid_token_responds_with_HTTP_200(self):
...
         self.assertEqual(response.status_code, 401)
+
+    # describe: healthcheck.json
+    def test_that_getting_healthcheck_json_with_a_valid_token_produces_parseable_json(self):
+        self.assert_produces_parseable_json(self.healthcheck(token=self.valid_token))
+
+    def test_that_when_the_database_is_ok_then_getting_healthcheck_json_with_a_valid_token_responds_with_HTTP_200(self):
+        with patch('postcode_api.views.HealthcheckDotJsonView.is_database_ok', return_value=True):
+            response = self.healthcheck()
+        self.assertEqual(response.status_code, 200)
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/tests/views/test_support_views.py]: You could have another assert method for this pattern too, eg:
```python
def assert_status_code(self, response, code):
    self.assertEqual(code, response.status_code)

def test_that_when_blah_HTTP_200(self):
    with patch('blah'):
        self.assert_status_code(self.healthcheck(), 200)
```

File: postcodeinfo/apps/postcode_api/tests/views/test_support_views.py [hunk 5]
```diff
@@ -25,16 +44,57 @@ def test_that_getting_ping_json_with_a_valid_token_responds_with_HTTP_200(self):
...
+
+    def test_that_getting_healthcheck_json_with_and_invalid_token_responds_with_HTTP_401(self):
+        response = self.healthcheck(token='Token 1234')
+        self.assertEqual(response.status_code, 401)
+
+    def test_that_when_the_database_is_ok_it_shows_ok_true_in_the_database_element(self):
+        with patch('postcode_api.views.HealthcheckDotJsonView.is_database_ok', return_value=True):
+            response = self.healthcheck(token=self.valid_token)
+        result = json.loads(response.content)
+        self.assertEqual( True, result['database']['ok'] )
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/tests/views/test_support_views.py]: And another assert for this pattern:
```python
def assert_json_ok(self, response, ok=False):
    self.assertEqual(ok, json.loads(response).content)['ok'])

def test_that_when_blah_ok_true(self):
    with patch('blah'):
        self.assert_json_ok(self.healthcheck(token=self.valid_token))
```

**Comment by andyhd** [comment]
:+1:

**Pull request merged**